google/cloud/aiplatform/_featurestores/entity_type.py

```diff
@@ -0,0 +1,456 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+import datetime
+import pandas as pd
+
+from google.auth import credentials as auth_credentials
+from google.protobuf import field_mask_pb2
+
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform.compat.types import entity_type as gca_entity_type
+from google.cloud.aiplatform import _featurestores
+from google.cloud.aiplatform import utils
+
+from google.cloud.aiplatform.utils import featurestore_utils
+
+_LOGGER = base.Logger(__name__)
+_ALL_FEATURE_IDS = "*"
+
+
+class EntityType(base.VertexAiResourceNounWithFutureManager):
+    """Managed entityType resource for Vertex AI."""
+
+    client_class = utils.FeaturestoreClientWithOverride
+
+    _is_client_prediction_client = False
+    _resource_noun = None
+    _getter_method = "get_entity_type"
+    _list_method = "list_entity_types"
+    _delete_method = "delete_entity_type"
+
+    def __init__(
+        self,
+        entity_type_name: str,
+        featurestore_id: Optional[str] = None,
```

Comment:
it's probably easier to keep the project , location, featurestore_id order

Reply:
One of the usage pattern for Vertex SDK is user to initialize the project, see usage example https://github.com/googleapis/python-aiplatform#initialization. And user does not need to specify `project` and `location` unless overwritten. That's why Vertex SDK normally ranks `project` and `location` to bottom among other arguments and mark as optional.

Also `Optional` arguments can not go before the `Required` arguments.